docs/cookbook.md

```diff
@@ -3,6 +3,54 @@ title:  Ion Schema Language Cookbook
 ---
 # {{ page.title }}
 
+## Optionally ignoring the occurs requirement for fields
+
+Suppose you have a type definition for a struct, and sometimes you want to validate the fields without enforcing the
+required-ness of fields. (The most likely use case for this is to perform partial validation on a partially constructed 
+piece of data.) Ion Schema does not provide any flag or option to "turn off" the required-ness of fields, but you can
+model this in Ion Schema by putting the types of fields and the required-ness of fields in separate type definitions.
+
+Here is an example using a `Customer` type:
+```ion
+type::{
+  name: CustomerAllFieldsOptional,
```

Comment:
CustomerBuilder? CustomerTemplate? CustomerSchema?

Reply:
I think all of those names have an implication that I want to avoid.

How about this? I think the names make it more clear what they are, and it's more concise than "CustomerAllFieldsOptional".
```
type::{
  name: CustomerFieldTypes,
  fields: {
    firstName: string,
    // ...
  },
  content: closed,
}

type::{
  name: CustomerFieldOccurrences,
  fields: {
    firstName: { occurs: required },
    // ...
  },
}

type::{
  name: Customer,
  all_of: [ CustomerFieldTypes, CustomerFieldOccurrence ]
}
```